### PR TITLE
[W06] Added PivotFinderTest 

### PR DESCRIPTION
**Pull request by mbmueller**
#### General

- [x] I confirm that my tests, if any, are correct
    - [x] Optional: I pass **all** artemis tests.
    - [x] Optional: I pass **all** tests, that are added in this PR.
    - [x] Optional: I calculated manually the correct solutions on paper.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://github.com/JohannesStoehr/gad23-tests/blob/main/CONTRIBUTING.md).
- [x] The newly added tests do not reveal the solutions dynamically calculating the solutions as expected in the exercise.
- [ ] Other students approved this Pull Request.

### Description
Added PivotFinderTest. Mainly added the Examples from Artemis, but also implemeted tests for Middle and Random Pivot as well as some Edge Cases for the Median Pivots.

<div style="width:100%;height:0;padding-bottom:83%;position:relative;"><iframe src="https://giphy.com/embed/UrbnbuU24p1zgyDMUH" width="100%" height="100%" style="position:absolute" frameBorder="0" class="giphy-embed" allowFullScreen></iframe></div><p><a href="https://giphy.com/gifs/Friends-season-5-episode-113-the-one-with-cop-UrbnbuU24p1zgyDMUH">via GIPHY</a></p>

### CODE REVIEW

**Review by Flnn0 — APPROVED** [review]
LGTM

File: w06/test/gad.simplesort/PivotFinderTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        numbers = new int[] {9, 1, 8, 5, 2, 3, 5, 1, 0, 7};
+        assertEquals(6, pivotFinder.findPivot(numbers, 0, 9));
+        assertEquals(5, pivotFinder.findPivot(numbers, 1, 9));
+    }
+
+    @Test
+    void testMedianPivotDistributed() {
+        PivotFinder pivotFinder = PivotFinder.getMedianPivotDistributed(5);
+        int[] numbers = new int[] {0, 1, 2, 3};
+        assertEquals(1, pivotFinder.findPivot(numbers, 0, 4));
```
Comment (SamuelScheit) [inline w06/test/gad.simplesort/PivotFinderTest.java:63]: testMedianPivotDistributed `from=0` `to=4` mit dem Array `new int[]{0, 1, 2, 3}` testet einen Index der außerhalb des Arrays liegt, weil das Array 4 Elemente groß ist und von index 0 bis index 3 läuft.
Nach der Artemis Aufgabenstellung sind `from` und `to` dabei beide inklusive.
D.h. der Test muss mit `from=0` und `to=3` testen.

> Wird der Median aus einer geraden Anzahl an Elementen gebildet, so ist das Ergebnis (normalerweise) der Durchschnitt der zwei mittleren Werte, hier kann aber einfach der kleinere mittlere Wert genutzt werden (wird nicht getestet).

`assertEquals(1, ...)` ist falsch, weil hier der Median 1 oder 2 sein darf.

**Review by SamuelScheit — CHANGES_REQUESTED** [review]

File: w06/test/gad.simplesort/PivotFinderTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
+package gad.simplesort;
```
Comment (Anatoly03) [inline w06/test/gad.simplesort/PivotFinderTest.java:1]:
```suggestion
package gad.bettersorting;
```

**Review by Anatoly03 — CHANGES_REQUESTED** [review]

File: w06/test/gad.simplesort/PivotFinderTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        assertEquals(5, pivotFinder.findPivot(numbers, 1, 9));
+    }
+
+    @Test
+    void testMedianPivotDistributed() {
+        PivotFinder pivotFinder = PivotFinder.getMedianPivotDistributed(5);
+        int[] numbers = new int[] {0, 1, 2, 3};
+        assertEquals(1, pivotFinder.findPivot(numbers, 0, 4));
+    }
+}
```
Comment (Anatoly03) [inline w06/test/gad.simplesort/PivotFinderTest.java:65]:
```suggestion
```

> Wird der Median aus einer geraden Anzahl an Elementen gebildet, so ist das [...] nicht getestet
